src/graphnet/data/i3extractor.py

```diff
@@ -327,36 +327,89 @@ def __call__(self, frame, padding_value=-1) -> dict:
             "SubEventID": frame["I3EventHeader"].sub_event_id,
+            "dbang_decay_length": padding_value,
+            "track_length": padding_value,
+            "stopped_muon": padding_value,
             "DeepCoreFilter_13": padding_value,
             "CascadeFilter_13": padding_value,
             "MuonFilter_13": padding_value,
             "OnlineL2Filter_17": padding_value,
-            "dbang_decay_length": padding_value,
-            "track_length": padding_value,
-            "stopped_muon": padding_value,
+            "L3_oscNext_bool": padding_value,
+            "L4_oscNext_bool": padding_value,
+            "L5_oscNext_bool": padding_value,
+            "L6_oscNext_bool": padding_value,
+            "L7_oscNext_bool": padding_value,
         }
 
         if (
             frame["I3EventHeader"].sub_event_stream == "InIceSplit"
         ):  # only inicesplit p frames have filters calculated
-            output.update(
-                {
-                    "DeepCoreFilter_13": int(
-                        bool(frame["FilterMask"]["DeepCoreFilter_13"])
-                    ),
-                    "CascadeFilter_13": int(
-                        bool(frame["FilterMask"]["CascadeFilter_13"])
-                    ),
-                    "MuonFilter_13": int(
-                        bool(frame["FilterMask"]["MuonFilter_13"])
-                    ),
-                    "OnlineL2Filter_17": int(
-                        bool(frame["FilterMask"]["OnlineL2Filter_17"])
-                    ),
-                }
-            )
 
-        if is_mc and (not is_noise):
+            if frame_has_key(frame, key="FilterMask"):
+                output["DeepCoreFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "DeepCoreFilter_13",
+                        default_value=-1,
+                    )
+                )  # what if frame doesnt have deepcorefilter and it says true (bool(-1)=true)
+                output["CascadeFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "CascadeFilter_13",
+                        default_value=-1,
+                    )
+                )
+                output["MuonFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"], "MuonFilter_13", default_value=-1
+                    )
+                )
+                output["OnlineL2Filter_17"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "OnlineL2Filter_17",
+                        default_value=-1,
+                    )
+                )
+
+            elif frame_has_key(frame, key="DeepCoreFilter_13"):
+                output["DeepCoreFilter_13"] = int(
+                    bool(frame["DeepCoreFilter_13"])
+                )
```

Comment:
Is the `int(bool(...))` conversion necessary, or will `int(...)` suffice?
```suggestion
                output["DeepCoreFilter_13"] = int(frame["DeepCoreFilter_13"])
```

Reply:
I think (though it has been a while since I tested) that the type of DeepCoreFilter_13 is I3Bool or something similar. This throws an error going straight to int, so needs the intermediate bool.